freebsd32 sync

brooksdavis opened the pull request:
A mostly-complete batch of commits syncing sys/compat/freebsd32/syscalls.master with sys/kern/syscalls.master. Includes a number of minor bug fixes found reviewing the diffs.

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -142,18 +143,17 @@
+52	AUE_SIGPENDING	COMPAT|NOPROTO	{ int sigpending(void); }
 53	AUE_SIGALTSTACK	STD	{ int freebsd32_sigaltstack( \
-				    struct sigaltstack32 *ss, \
+				    const struct sigaltstack32 *ss, \
```
This one doesn't seem to be const in !freebsd32, but it would seem reasonable to apply it there (and the kern_* helper); same for a couple more down below.

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -191,164 +190,174 @@
 82	AUE_SETPGRP	NOPROTO	{ int setpgid(int pid, int pgid); }
-83	AUE_SETITIMER	STD	{ int freebsd32_setitimer(u_int which, \
-				    struct itimerval32 *itv, \
+83	AUE_SETITIMER	STD	{ int freebsd32_setitimer(int which, \
```
!freebsd32 is u_int; ditto for getitimer

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -191,164 +190,174 @@
 				    const struct sockaddr *name, \
-				    int namelen); }
-99	AUE_NULL	OBSOL	oaccept
+				    __socklen_t namelen); }
```
!freebsd32 is an int and probably shouldn't be

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -191,164 +190,174 @@
 				    struct ia32_sigcontext3 *sigcntxp); }
 104	AUE_BIND	NOPROTO	{ int bind(int s, const struct sockaddr *name, \
-				    int namelen); }
+				    __socklen_t namelen); }
```
Ditto (!freebsd32 should be socklen_t)

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -191,164 +190,174 @@
+				    __socklen_t namelen); }
 105	AUE_SETSOCKOPT	NOPROTO	{ int setsockopt(int s, int level, \
-				    int name, const void *val, int valsize); }
+				    int name, const void *val, __socklen_t valsize); }
```
Ditto

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -191,164 +190,174 @@
-112	AUE_O_SIGSTACK	COMPAT	{ int freebsd32_sigstack( \
+109	AUE_O_SIGBLOCK	COMPAT|NOPROTO	{ int sigblock(int mask); }
+110	AUE_O_SIGSETMASK	COMPAT|NOPROTO	{ int sigsetmask(int mask); }
+111	AUE_SIGSUSPEND	COMPAT|NOPROTO	{ int sigsuspend( int mask); }
```
mask is pedantically an `osigset_t`

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -191,164 +190,174 @@
-114	AUE_NULL	OBSOL	osendmsg
+113	AUE_RECVMSG	COMPAT	{ int freebsd32_recvmsg(int s, \
+				    struct omsghdr32 *msg, int flags); }
+114	AUE_SENDMSG	OBSOL	osendmsg
```
I note that osendmsg is also not disabled in !freebsd32

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -191,164 +190,174 @@
 118	AUE_GETSOCKOPT	NOPROTO	{ int getsockopt(int s, int level, \
-				    int name, void *val, int *avalsize); }
-119	AUE_NULL	UNIMPL	resuba (BSD/OS 2.x)
+				    int name, void *val, __socklen_t *avalsize); }
```
Ditto previous comment -- !freebsd32 counterpart should probably be socklen_t

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -491,60 +500,73 @@
-270	AUE_NULL	UNIMPL	nosys
-271	AUE_NULL	UNIMPL	nosys
-272	AUE_O_GETDENTS	COMPAT11 { int freebsd32_getdents(int fd, char *buf, \
+258	AUE_NULL	RESERVED
```
We might collapse these to the range syntax for symmetry in syscalls.master pre-commit (noted that it is convenient for the review, indeed :-))

kevans91 commented on sys/compat/freebsd32/syscalls.master:
```diff
@@ -491,60 +500,73 @@
 				    struct aiocb32 *aiocbp); }
 257	AUE_LIO_LISTIO	STD	{ int freebsd32_lio_listio(int mode, \
-				    struct aiocb32 * const *acb_list, \
+				    uint32_t *acb_list, \
```
Is this also one of your pending makesyscalls.lua changes, or one we'll still need to make?

kevans91 reviewed (commented):
Noted a couple things I'm unsure of, but it's likely that I just missed something. :-)

brooksdavis commented:
I think I've addressed the identified issues except for `osigset_t` which I'd like to leave for later.

kevans91 commented:
Changes LGTM, and no objection to punting on the bits you've elected to punt on. Thanks!

brooksdavis commented:
Rebased and updated so generated files are included in the commits that change them. This aids bisection since every commit should be buildable. I've moved the makesyscalls.lua and general syscall bits up to the top. I also dropped the jail argument rename as it's not in sys/kern/syscalls.master (I'll merge the whole batch of not naming syscall arguments the same thing as the syscall later since it's not needed now.)

brooksdavis commented:
Merged as 1739de97..572be9c8
